Module 5/Question 15/Problem15.json

```diff
@@ -0,0 +1,188 @@
+Input Json
+
+{
+    "orders": [
+        {
+            "id": "order1",
+            "date": "2022-01-01T00:00:00Z",
+            "customerId": "customer1",
+            "items": [
+                {
+                    "id": "item1",
+                    "name": "Product 1",
+                    "price": 10,
+                    "quantity": 2
+                },
+                {
+                    "id": "item2",
+                    "name": "Product 2",
+                    "price": 20,
+                    "quantity": 1
+                }
+            ],
+            "totalPrice": 40
+        },
+        {
+            "id": "order2",
+            "date": "2022-02-01T00:00:00Z",
+            "customerId": "customer1",
+            "items": [
+                {
+                    "id": "item3",
+                    "name": "Product 3",
+                    "price": 30,
+                    "quantity": 3
+                }
+            ],
+            "totalPrice": 90
+        },
+        {
+            "id": "order3",
+            "date": "2022-03-01T00:00:00Z",
+            "customerId": "customer2",
+            "items": [
+                {
+                    "id": "item4",
+                    "name": "Product 4",
+                    "price": 40,
+                    "quantity": 4
+                }
+            ],
+            "totalPrice": 160
+        }
+    ]
+}
+
+Json Spec
+
+[
+    {
+        "operation": "shift",
+        "spec": {
+            "orders": {
+                "*": {
+                    "@": "@(1,customerId).orders[]"
+                }
+            }
+        }
+    },
+    {
+        "operation": "shift",
+        "spec": {
+            "*": "customers"
+        }
+    },
+    {
+        "operation": "shift",
+        "spec": {
+            "customers": {
+                "*": {
+                    "orders": {
+                        "*": {
+                            "*": "customers[#4].orders[#2].&",
+                            "totalPrice": "customers[#4].totalAmount"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    {
+        "operation": "modify-overwrite-beta",
+        "spec": {
+            "customers": {
+                "*": {
+                    "totalAmount": "=intSum(@(1,totalAmount))",
+                    "id": "@(1,orders[0].customerId)"
+                },
+                "[0]": {
```

Comment:
totalPrice field is missing in output also using customerId instead of index to add name will be better.